### PR TITLE
chore: refactor connect into connect_info

### LINKED ISSUE #290: Refactor "connect" into "connect_info"
Update "ICM.connect" to "ICM.connect_info"

This should also move the "connect_with_XXXX" functions up to the connector level as well.

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Closes #290

### CODE REVIEW

File: google/cloud/sql/connector/instance_connection_manager.py [hunk 1]
```diff
@@ -438,171 +429,31 @@ async def _refresh_task(
...
-                raise PlatformNotSupportedError(
-                    "Active Directory authentication is currently only supported on Windows."
-                )
-
-        user = kwargs.pop("user")
-        passwd = kwargs.pop("password")
-        return pytds.connect(
-            ip_address, database=db, user=user, password=passwd, sock=sock, **kwargs
-        )
+        return (instance_data, ip_address)
```
Comment (kurtisvg) [inline google/cloud/sql/connector/instance_connection_manager.py:456]: nit: You don't need to explicitly bundle this in a tuple, python does it automatically with just `return a, b`

File: google/cloud/sql/connector/utils.py [hunk 2]
```diff
@@ -77,3 +80,27 @@ def write_to_file(
         priv_out.write(priv_key)
 
     return (ca_filename, cert_filename, key_filename)
+
+
+def clean_input(default_ip_type: Any, default_timeout: int, **kwargs: Any) -> Any:
```
Comment (kurtisvg) [inline google/cloud/sql/connector/utils.py:85]: We're only using this in the 1 place - is it really worth separating out into it's own function?

File: google/cloud/sql/connector/connector.py [hunk 3]
```diff
@@ -229,3 +243,169 @@ def connect(instance_connection_string: str, driver: str, **kwargs: Any) -> Any:
...
+        else:
+            raise PlatformNotSupportedError(
+                "Active Directory authentication is currently only supported on Windows."
+            )
+
+    user = kwargs.pop("user")
+    passwd = kwargs.pop("password")
+    return pytds.connect(
+        ip_address, database=db, user=user, password=passwd, sock=sock, **kwargs
+    )
```
Comment (kurtisvg) [inline google/cloud/sql/connector/connector.py:411]: nit: these 3 functions might be easier to find if we move the into their own files

File: google/cloud/sql/connector/connector.py [hunk 4]
```diff
@@ -229,3 +243,169 @@ def connect(instance_connection_string: str, driver: str, **kwargs: Any) -> Any:
...
     return _default_connector.connect(instance_connection_string, driver, **kwargs)
+
+
+async def get_connection(
+    icm: InstanceConnectionManager,
+    connect_function: Callable,
+    ip_type: IPTypes,
+    loop: asyncio.AbstractEventLoop,
+    **kwargs: Any,
+) -> Any:
```
Comment (kurtisvg) [inline google/cloud/sql/connector/connector.py:254]: I don't really understand this function's signature (or why we need it). It's also very short (minus the docstring). Are we sure we shouldn't inline here?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: google/cloud/sql/connector/utils.py [hunk 5]
```diff
@@ -19,6 +19,9 @@
 from cryptography.hazmat.primitives.asymmetric import rsa
 
 from typing import Tuple
+import logging
+
+logger = logging.getLogger(name=__name__)
```
Comment (kurtisvg) [inline google/cloud/sql/connector/utils.py:24]: Are we using these here?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: google/cloud/sql/connector/connector.py [hunk 6]
```diff
@@ -170,10 +186,31 @@ async def connect_async(
...
+        kwargs.pop("ssl", None)
+        kwargs.pop("port", None)
+
+        # helper function to wrap in timeout
+        async def get_connection(
+            self: Connector,
+            icm: InstanceConnectionManager,
+            connect_function: Callable,
+            ip_type: IPTypes,
+            **kwargs: Any,
```
[the remark this answers is not in the document]
Comment (jackwotherspoon) [reply google/cloud/sql/connector/connector.py]: Removed need for these arguments.

File: google/cloud/sql/connector/connector.py [hunk 7]
```diff
@@ -171,11 +187,24 @@ async def connect_async(
...
 
+        # Host and ssl options come from the certificates and metadata, so we don't
+        # want the user to specify them.
+        kwargs.pop("host", None)
+        kwargs.pop("ssl", None)
+        kwargs.pop("port", None)
+
+        # helper function to wrap in timeout
+        async def get_connection() -> Any:
+
```
Comment (kurtisvg) [inline google/cloud/sql/connector/connector.py:198]: nit: 
```suggestion
```

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**